File: src/ffxiv/jobs.rs
use std::collections::HashMap;
use ffxiv_types::jobs::{ClassJob, Class, Job, NonCombatJob};

lazy_static::lazy_static! {
    pub static ref JOBS: HashMap<u32, ClassJob> = maplit::hashmap! {
        1 => ClassJob::Class(Class::Gladiator),
        2 => ClassJob::Class(Class::Pugilist),
        3 => ClassJob::Class(Class::Marauder),
        4 => ClassJob::Class(Class::Lancer),
        5 => ClassJob::Class(Class::Archer),
        6 => ClassJob::Class(Class::Conjurer),
        7 => ClassJob::Class(Class::Thaumaturge),
        8 => ClassJob::NonCombat(NonCombatJob::Carpenter),
        9 => ClassJob::NonCombat(NonCombatJob::Blacksmith),
        10 => ClassJob::NonCombat(NonCombatJob::Armorer),
        11 => ClassJob::NonCombat(NonCombatJob::Goldsmith),
        12 => ClassJob::NonCombat(NonCombatJob::Leatherworker),
        13 => ClassJob::NonCombat(NonCombatJob::Weaver),
        14 => ClassJob::NonCombat(NonCombatJob::Alchemist),
        15 => ClassJob::NonCombat(NonCombatJob::Culinarian),
        16 => ClassJob::NonCombat(NonCombatJob::Miner),
        17 => ClassJob::NonCombat(NonCombatJob::Botanist),
        18 => ClassJob::NonCombat(NonCombatJob::Fisher),
        19 => ClassJob::Job(Job::Paladin),
        20 => ClassJob::Job(Job::Monk),
        21 => ClassJob::Job(Job::Warrior),
        22 => ClassJob::Job(Job::Dragoon),
        23 => ClassJob::Job(Job::Bard),
        24 => ClassJob::Job(Job::WhiteMage),
        25 => ClassJob::Job(Job::BlackMage),
        26 => ClassJob::Class(Class::Arcanist),
        27 => ClassJob::Job(Job::Summoner),
        28 => ClassJob::Job(Job::Scholar),
        29 => ClassJob::Class(Class::Rogue),
        30 => ClassJob::Job(Job::Ninja),
        31 => ClassJob::Job(Job::Machinist),
        32 => ClassJob::Job(Job::DarkKnight),
        33 => ClassJob::Job(Job::Astrologian),
        34 => ClassJob::Job(Job::Samurai),
        35 => ClassJob::Job(Job::RedMage),
        36 => ClassJob::Job(Job::BlueMage),
        37 => ClassJob::Job(Job::Gunbreaker),
        38 => ClassJob::Job(Job::Dancer),
    };
}

File: csharp/SourceGenerator/preview/ffxiv/jobs.go
package ffxiv

type ClassJob interface {
	isClassJob()
}

type Class string

func (Class) isClassJob() {}

type Job string

func (Job) isClassJob() {}

type NonCombatJob string

func (NonCombatJob) isClassJob() {}

const (
	Gladiator   Class = "Gladiator"
	Pugilist    Class = "Pugilist"
	Marauder    Class = "Marauder"
	Lancer      Class = "Lancer"
	Archer      Class = "Archer"
	Conjurer    Class = "Conjurer"
	Thaumaturge Class = "Thaumaturge"
	Arcanist    Class = "Arcanist"
	Rogue       Class = "Rogue"
)

const (
	Paladin     Job = "Paladin"
	Monk        Job = "Monk"
	Warrior     Job = "Warrior"
	Dragoon     Job = "Dragoon"
	Bard        Job = "Bard"
	WhiteMage   Job = "WhiteMage"
	BlackMage   Job = "BlackMage"
	Summoner    Job = "Summoner"
	Scholar     Job = "Scholar"
	Ninja       Job = "Ninja"
	Machinist   Job = "Machinist"
	DarkKnight  Job = "DarkKnight"
	Astrologian Job = "Astrologian"
	Samurai     Job = "Samurai"
	RedMage     Job = "RedMage"
	BlueMage    Job = "BlueMage"
	Gunbreaker  Job = "Gunbreaker"
	Dancer      Job = "Dancer"
	Reaper      Job = "Reaper"
	Sage        Job = "Sage"
	Viper       Job = "Viper"
	Pictomancer Job = "Pictomancer"
)

const (
	Carpenter     NonCombatJob = "Carpenter"
	Blacksmith    NonCombatJob = "Blacksmith"
	Armorer       NonCombatJob = "Armorer"
	Goldsmith     NonCombatJob = "Goldsmith"
	Leatherworker NonCombatJob = "Leatherworker"
	Weaver        NonCombatJob = "Weaver"
	Alchemist     NonCombatJob = "Alchemist"
	Culinarian    NonCombatJob = "Culinarian"
	Miner         NonCombatJob = "Miner"
	Botanist      NonCombatJob = "Botanist"
	Fisher        NonCombatJob = "Fisher"
)

var JOBS = map[uint32]ClassJob{
	1:  Gladiator,
	2:  Pugilist,
	3:  Marauder,
	4:  Lancer,
	5:  Archer,
	6:  Conjurer,
	7:  Thaumaturge,
	8:  Carpenter,
	9:  Blacksmith,
	10: Armorer,
	11: Goldsmith,
	12: Leatherworker,
	13: Weaver,
	14: Alchemist,
	15: Culinarian,
	16: Miner,
	17: Botanist,
	18: Fisher,
	19: Paladin,
	20: Monk,
	21: Warrior,
	22: Dragoon,
	23: Bard,
	24: WhiteMage,
	25: BlackMage,
	26: Arcanist,
	27: Summoner,
	28: Scholar,
	29: Rogue,
	30: Ninja,
	31: Machinist,
	32: DarkKnight,
	33: Astrologian,
	34: Samurai,
	35: RedMage,
	36: BlueMage,
	37: Gunbreaker,
	38: Dancer,
	39: Reaper,
	40: Sage,
	41: Viper,
	42: Pictomancer,
}
